React/Modules/RCTAlertManager.m

```diff
@@ -155,13 +176,20 @@ - (void)invalidate
       NSString *buttonKey = button.allKeys.firstObject;
       NSString *buttonTitle = [button[buttonKey] description];
       UIAlertActionStyle buttonStyle = [buttonKey isEqualToString:@"cancel"] ? UIAlertActionStyleCancel : UIAlertActionStyleDefault;
-      UITextField *textField = allowsTextInput ? alertController.textFields.firstObject : nil;
       [alertController addAction:[UIAlertAction actionWithTitle:buttonTitle
                                                           style:buttonStyle
                                                         handler:^(__unused UIAlertAction *action) {
         if (callback) {
-          if (allowsTextInput) {
-            callback(@[buttonKey, textField.text]);
+          if (type.length) {
```

Comment:
as in

    if (type plain text or secure) {

    } else if (type login-password) {

    } else {

    }